server/channels/app/platform/web_conn.go

```diff
@@ -831,7 +833,16 @@ func (wc *WebConn) ShouldSendEvent(msg *model.WebSocketEvent) bool {
 	}
 
 	// Only report events to users who are in the channel for the event
-	if msg.GetBroadcast().ChannelId != "" {
+	if chID := msg.GetBroadcast().ChannelId; chID != "" {
+		// For typing events, we don't send them to users who don't have
+		// that channel or thread opened.
+		if msg.EventType() == model.WebsocketEventTyping {
+			if (wc.GetActiveChannelID() != "<>" && chID != wc.GetActiveChannelID()) &&
+				(wc.GetActiveThreadChannelID() != "<>" && chID != wc.GetActiveThreadChannelID()) {
+				return false
+			}
```

Comment:
Why the parenthesis when they are all and conditions? 

Also wondering if making this a function would simplify a bit, especially as I could see this reused for other events.

Reply:
Sure